src/connection.rs

```diff
@@ -88,7 +88,7 @@ pub trait RequestConnection {
         fds: Vec<RawFdContainer>,
     ) -> Result<Cookie<'_, Self, R>, ConnectionError>
     where
-        R: TryFrom<Buffer, Error = ParseError>;
+        R: for<'a> TryFrom<&'a [u8], Error = ParseError>;
```

Comment:
Okay, this is something that I would never have come up with. Today I learned about what `for<'a>` is good for... (I saw this before, but this use here makes a lot of sense to me).

Thanks!

Reply:
Unfortunately, I could not get around the `single_use_lifetimes` lint, so I disabled it.